docker-compose-files/compose.sh

```diff
@@ -229,16 +191,79 @@ echo "GROUP_ID=$(id -g)" >> $topdir/.env
 echo "DOCKER_INFLUXDB_INIT_ADMIN_TOKEN=${DOCKER_INFLUXDB_INIT_ADMIN_TOKEN}" >> $topdir/.env
 echo "DOCKER_INFLUXDB_INIT_PASSWORD=${DOCKER_INFLUXDB_INIT_PASSWORD}" >> $topdir/.env
 
+# init Splink env variables if not set to avoid warnings from docker-compose for other pumps
+if [ -z $SPLUNK_HEC_URL ]; then
+   export SPLUNK_HEC_URL=
+fi
+if [ -z $SPLUNK_HEC_KEY ]; then
+   export SPLUNK_HEC_KEY=
+fi
+if [ -z $SPLUNK_HEC_INDEX ]; then
+   export SPLUNK_HEC_INDEX=
+fi
+
+ # remove dependency on setup influx-test-db
+touch $topdir/docker-compose-files/container-info-influx-pump.txt 
+touch $topdir/docker-compose-files/container-info-grafana.txt
+
 case $1 in
   rm)
     docker-compose --project-directory $topdir -f $scriptdir/docker-compose.yml ${PROFILE_ARG} rm -f
     ;;
 
+  setup)
+      if [[ -z $INFLUX ]]; then
+          echo  "Setup only with influx-test-db  profile... "
+          exit 1
+      fi
+      # RUN ONLY setup containers
+      PROFILE_ARG="--profile setup-influx-test-db"
+      POST_ACTION="influx_setup_finish"
+      DETACH_ARG="-d"
+      BUILD_ARG=
+      #eval set -- "start"
+
+      for i in  $topdir/docker-compose-files/container-info-influx-pump.txt $topdir/docker-compose-files/container-info-grafana.txt
+      do
+        rm -f $i
+        touch $i
+      done
+
+      # delete any older setup for grafana
+      for container in telemetry-reference-tools-influx telemetry-reference-tools-grafana
+      do
+        id=$(docker container ls -a --filter name=$container -q)
+        if [[ -z $id ]]; then
+          continue
+        fi
+        echo "Cleaning up old containers for $container: $id"
+        echo -n "Stopping: "
+        docker container stop $id
+        echo -n "Removing: "
+        docker container rm -v $id
+      done
+
+      volume=$(docker volume ls --filter name=telemetry-reference-tools_influxdb-storage -q)
+      if [[ -n $volume ]]; then
+        echo -n "Removing volume: $volume"
+        docker volume rm $volume
+      fi
+      docker-compose --project-directory $topdir -f $scriptdir/docker-compose.yml ${PROFILE_ARG} up ${BUILD_ARG} ${DETACH_ARG}
+      ;;  
+
   stop)
     docker-compose --project-directory $topdir -f $scriptdir/docker-compose.yml ${PROFILE_ARG} stop
     ;;
 
-  start)
+  start)  
+    if [[ -n $SPLUNK ]] && [[ -z $SPLUNK_HEC_URL || -z $SPLUNK_HEC_KEY || -z $SPLUNK_HEC_INDEX ]];then 
+       echo "Splunk env variables SPLUNK_HEC_URL, SPLUNK_HEC_KEY, and/or SPLUNK_HEC_INDEX not set! "
+       exit 1
+    fi
+    if  [[ -n $INFLUX ]] && [[ ! -s docker-compose-files/container-info-influx-pump.txt ]]; then
```

Comment:
This test wont work because it is touching this file unconditionally above.

Reply:
Ok, I see that now. The old code was doing "-e" which tests for existence and I didnt see you changed it to a "-s"